core/src/consensus/solo/mod.rs

```diff
@@ -81,23 +81,23 @@ impl ConsensusEngine for Solo<CodeChainMachine> {
 
     fn on_close_block(&self, block: &mut ExecutedBlock, parent_common_params: &CommonParams) -> Result<(), Error> {
         let author = *block.header().author();
-        let (total_reward, min_fee) = {
+        let (total_reward, total_min_fee) = {
             let transactions = block.transactions();
             let block_reward = self.block_reward(block.header().number());
-            let total_fee: u64 = transactions.iter().map(|tx| tx.fee).sum();
+            let total_min_fee: u64 = transactions.iter().map(|tx| tx.fee).sum();
             let min_fee: u64 =
                 transactions.iter().map(|tx| CodeChainMachine::min_cost(&parent_common_params, &tx.action)).sum();
-            (block_reward + total_fee, min_fee)
+            (block_reward + total_min_fee, min_fee)
         };
 
-        assert!(total_reward >= min_fee, "{} >= {}", total_reward, min_fee);
+        assert!(total_reward >= total_min_fee, "{} >= {}", total_reward, total_min_fee);
         let stakes = stake::get_stakes(block.state()).expect("Cannot get Stake status");
-        for (address, share) in stake::fee_distribute(&author, min_fee, &stakes) {
+        for (address, share) in stake::fee_distribute(total_min_fee, &stakes) {
             self.machine.add_balance(block, &address, share)?
         }
-        if total_reward != min_fee {
-            self.machine.add_balance(block, &author, total_reward - min_fee)?
-        }
+        let stakeholders_share = stake::stakeholders_share(total_min_fee, &stakes);
+        self.machine.add_balance(block, &author, total_reward - stakeholders_share)?;
```

Comment:
Why iterate stakes twice?
Is this more clear to express the intention that "remaining fee due to the rounding goes to the author"
```rust
let mut distributor = stake::fee_distribute(total_min_fee, &stakes);
for (address, share) in distributor {
  ...
}
self.mahcine.add_balance(block, &author, total_reward - total_min_fee + distributor.get_remaining_fee())?;
```

Reply:
Because the block author doesn't take the fee at block close after term change introduced.

Reply:
Will it be greatly different from this?
```rust
stake.add_pending_reward(&author, total_reward - total_min_fee + distributor.get_remaining_fee());
```

Reply:
Two things are the same because `stakeholders_share` uses the iteration internally.
But when we move the fee distribution to term change, there will be no iterators, so reusing the iterator doesn't have merit at all.

Reply:
I meant that it 'iterates' HashMap twice, not meant that it doesn't 'reuse iterator'. Anyway, it is a minor thing. I'll approve this PR.

Reply:
I meant you cannot make it iterate once.
From the next patch, you cannot call `add_pending_reward` here, you should recalculate the reward of every block that generates in a term.

Reply:
The pending_reward is a field that the former validators will receive.
If you want to store the block reward in a term to avoid calculating it twice, you should introduce a new field.

Reply:
I think it has merit to store the intermediate reward in states.
```
intermediate_rewards = [ [ address, quantity ] * ]
```
@foriequal0 How about you?